File: android/src/main/kotlin/com/tsitser/background_http_plugin/domain/usecase/CancelRequestUseCase.kt
package com.tsitser.background_http_plugin.domain.usecase

import com.tsitser.background_http_plugin.domain.repository.TaskRepository

/**
 * Use case для отмены задачи
 */
class CancelRequestUseCase(
    private val repository: TaskRepository
) {
    suspend operator fun invoke(requestId: String): Boolean? {
        return repository.cancelTask(requestId)
    }
}

File: ios/Classes/Domain/UseCase/CancelRequestUseCase.swift
import Foundation

/// Use case for canceling a task
struct CancelRequestUseCase {
    private let repository: TaskRepository
    
    init(repository: TaskRepository) {
        self.repository = repository
    }
    
    func execute(requestId: String) async throws -> Bool? {
        return try await repository.cancelTask(requestId: requestId)
    }
}

